selfservice/strategy/profile/strategy_test.go

```diff
@@ -309,30 +317,31 @@ func TestStrategyTraits(t *testing.T) {
 		}
 
 		t.Run("type=api", func(t *testing.T) {
-			setUnprivileged(t)
+			apiUser1 := getApiUser(false, apiIdentity1)
 			f := testhelpers.InitializeSettingsFlowViaAPI(t, apiUser1, publicTS)
 			res := run(t, f, true, apiUser1)
 			assert.EqualValues(t, http.StatusForbidden, res.StatusCode)
 			assert.Contains(t, res.Request.URL.String(), publicTS.URL+settings.RouteSubmitFlow)
 		})
 
 		t.Run("type=sqa", func(t *testing.T) {
-			setUnprivileged(t)
+			browserUser1 := getBrowserUser(false, browserIdentity1)
 			f := testhelpers.InitializeSettingsFlowViaBrowser(t, browserUser1, true, publicTS)
 			res := run(t, f, true, browserUser1)
 			assert.EqualValues(t, http.StatusUnauthorized, res.StatusCode)
 			assert.Contains(t, res.Request.URL.String(), conf.GetProvider(ctx).String(config.ViperKeySelfServiceLoginUI))
 		})
 
 		t.Run("type=browser", func(t *testing.T) {
-			setUnprivileged(t)
+			browserUser1 := getBrowserUser(false, browserIdentity1)
 			f := testhelpers.InitializeSettingsFlowViaBrowser(t, browserUser1, false, publicTS)
 			res := run(t, f, false, browserUser1)
 			assert.EqualValues(t, http.StatusUnauthorized, res.StatusCode)
 			assert.Contains(t, res.Request.URL.String(), conf.GetProvider(ctx).String(config.ViperKeySelfServiceLoginUI))
 
 			t.Run("should update when signed back in", func(t *testing.T) {
-				setPrivileged(t)
+				browserUser1 := getBrowserUser(true, browserIdentity1)
+				_ = run(t, f, false, browserUser1)
```

Comment:
IIRC there's a side effect in the run function that's needed for the `browserUser1.get(f.Ui.Action)` on the next line to work correctly. This "solution" of creating a new client and then essentially re-doing the first part of this test (from line 338) is a little hacky, and difficult to understand in the long run.

I wrote this before I added the `RecordSessionMutator` functionality to the session test helpers. This test should be changed to use that session mutator and just change the session to privileged here rather than re-creating a client